backend/src/services/comment/comment.service.ts

```diff
@@ -1,23 +1,36 @@
 import { Comment as TComment, CommentCreatePayload } from '~/common/types/types';
-import { comment as commentRep } from '~/data/repositories/repositories';
+import {
+  comment as commentRep,
+  user as userRep,
+} from '~/data/repositories/repositories';
 
 type Constructor = {
   commentRepository: typeof commentRep;
+  userRepository: typeof userRep;
 };
 
 class Comment {
   #commentRepository: typeof commentRep;
+  #userRepository: typeof userRep;
 
-  constructor({ commentRepository }: Constructor) {
+  constructor({ commentRepository, userRepository }: Constructor) {
     this.#commentRepository = commentRepository;
+    this.#userRepository = userRepository;
   }
 
   public getAll(): Promise<TComment[]> {
     return this.#commentRepository.getAll();
   }
 
-  public create(payload: CommentCreatePayload): Promise<TComment> {
-    return this.#commentRepository.create(payload);
+  public async create(payload: CommentCreatePayload): Promise<TComment> {
+    const user = await this.#userRepository.getById(payload.userId);
+    const newComment = { ...payload, user };
```

Comment:
you already add `withGraphFetched `, so now you don't need to do it 😉
Model does everything!